packages/conversation/src/rest/v1/conversation-domain-api.ts

```diff
@@ -59,9 +61,13 @@ export class ConversationDomainApi implements Api {
    */
   public getSinchClient(): ApiClient {
     if (!this.client) {
+      const region = this.sinchClientParameters.region || Region.UNITED_STATES;
+      if(!Object.values(ConversationRegion).includes((region as unknown) as ConversationRegion)) {
+        throw new Error(`The region '${region}' is not supported for the Conversation API`);
+      }
       const apiClientOptions = this.buildApiClientOptions(this.sinchClientParameters);
       this.client = new ApiFetchClient(apiClientOptions);
-      this.client.apiClientOptions.basePath = 'https://us.conversation.api.sinch.com';
+      this.client.apiClientOptions.basePath = `https://${region}.conversation.api.sinch.com`;
```

Comment:
Then currently the region is only to set the URL, but for internal or external validation purpose are we able to set a specific URL value ?
e.g. How to use `https:/foo.com` in place of `https://${region}.conversation.api.sinch.com`
- the value is set by end user
- there is not format to apply in regard of region value

Reply:
The method `setBasePath()` is there for that :) 
I've added a test to illustrate its usage.

Reply:
ok. overriding the value have to be done after initialization
This emphasizes that the exception must not be raised during initialization 👍